src/Security/SslConnection.cs

```diff
@@ -69,25 +75,38 @@ protected virtual void Dispose (bool disposing)
 
 		public IntPtr ConnectionId { get; private set; }
 
+#if NET
+		unsafe internal delegate* unmanaged<IntPtr, IntPtr, nint*, SslStatus> ReadFunc { get { return &Read; } }
+		unsafe internal delegate* unmanaged<IntPtr, IntPtr, nint*, SslStatus> WriteFunc { get { return &Write; } }
+#else
 		internal SslReadFunc ReadFunc { get; private set; }
 		internal SslWriteFunc WriteFunc { get; private set; }
+#endif
 
 		public abstract SslStatus Read (IntPtr data, ref nint dataLength);
 
 		public abstract SslStatus Write (IntPtr data, ref nint dataLength);
 
+#if NET
+		[UnmanagedCallersOnly]
+#else
 		[MonoPInvokeCallback (typeof (SslReadFunc))]
-		static SslStatus Read (IntPtr connection, IntPtr data, ref nint dataLength)
+#endif
+		unsafe static SslStatus Read (IntPtr connection, IntPtr data, nint* dataLength)
 		{
 			var c = (SslConnection) GCHandle.FromIntPtr (connection).Target;
-			return c.Read (data, ref dataLength);
+			return c.Read (data, ref System.Runtime.CompilerServices.Unsafe.AsRef<nint> (dataLength));
```

Comment:
Why the pointer usafe here? It's possible to use `ref nint` in the signature.

Reply:
@Therzok no, it's not possible to use `ref nint` in the signature, because it's not blittable: https://github.com/dotnet/roslyn/issues/57025